solidity/dashboard/src/pages/rewards/RewardsOverviewPage.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect } from "react"
+import { useDispatch, useSelector } from "react-redux"
+import TokenAmount from "../../components/TokenAmount"
+// import { SubmitButton } from "../../components/Button"
+import BeaconRewardsDetails from "../../components/BeaconRewardsDetails"
+import StakeDropChart from "../../components/StakeDropChart"
+import { useWeb3Context } from "../../components/WithWeb3Context"
+import { TokenAmountSkeleton } from "../../components/skeletons"
+
+const RewardsOverviewPage = () => {
+  const dispatch = useDispatch()
+  const { yourAddress } = useWeb3Context()
+  const { beaconRewardsFetching, becaonRewardsBalance } = useSelector(
+    (state) => state.rewards
+  )
+
+  useEffect(() => {
+    dispatch({
+      type: "rewards/beacon_fetch_distributed_rewards_request",
+      payload: yourAddress,
```

Comment:
Any chance we can at least make this work for the token holder as well, not just the beneficiary address? I actually think that might be a blocker, and it may make more sense to make this _only_ work for the token holder than to have it _only_ work for the beneficiary for this initial version.

Reply:
It's probably doable but probably lots of extra calls will be needed since rewards are transferred to a `beneficiary` and the beneficiary may have more than one operator. Also would have to call `getGroupMembers` and check if a given operator address is in a given group. The next step is computing a reward per operator. My point is that we don't have a direct match between transferred tokens from rewards and a given operator address. Probably it won't be very complicated, just a little bit tricky I guess.

Also, Rewards from `KeepRandomBeaconOperator` contract are displayed from the beneficiary's perspective.

Reply:
A'ight, I think we can stick with what we've got for now.